src/main/java/com/epam/java/pdp/figures/Shape.java

```diff
@@ -18,44 +18,53 @@ public Shape(Point... points) {
 
     boolean isTriangle() {
         if (points.size() == 3) {
-            System.out.println("This is triangle");
+//            calculateLineLengths();
+            arePointsOnTheSingleLine();
+            System.out.println("This is a triangle");
         }
         return true;
+
     }
 
 
-    double calculateLines() {
-        this.lines = new ArrayList<>();
-        for (int i = 0; i < points.size(); ++i) {
-            if (i != points.size()) {
-                Line ln = (this.points.get(i), this.points.get(i++));
-                this.lines.add(ln.lineLength((this.points.get(i)), this.points.get(i++)));
-            } else {
-                Line ln = (this.points.get(i), this.points.get(0));
-                this.lines.add(ln.lineLength((this.points.get(i)), this.points.get(i++)));
+    void calculateLineLengths() {
+        this.lineLengths = new ArrayList<>();
+        for (int i = 0; i < points.size(); i++) {
+            Point start = points.get(i);
+            int endIndex = i + 1;
+            if (endIndex == points.size()) {
+                endIndex = 0;
             }
+            Point end = this.points.get(endIndex);
+            Line line = new Line(start, end);
+            double length = line.lineLength();
+            lineLengths.add(length);
+
         }
 
     }
 
-    double areLinesOnTheSingleLine() {
+    void arePointsOnTheSingleLine() {
```

Comment:
What should be the desired output/return type of that method?

Reply:
added variable oneLine to return true if points are on the same line